nncf/experimental/quantization/quantizer/torch_ao_adapter.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) 2025 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from collections import defaultdict
+from copy import deepcopy
+from typing import Dict, List, Tuple, Union
+
+import torch
+import torch.fx
+from torch.ao.quantization.quantizer import Quantizer
+from torch.ao.quantization.quantizer.quantizer import QuantizationSpec
+from torch.ao.quantization.quantizer.quantizer import QuantizationSpecBase
+from torch.ao.quantization.quantizer.quantizer import SharedQuantizationSpec
+
+import nncf
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.logging import nncf_logger
+from nncf.common.quantization.quantizer_setup import ActivationQuantizationInsertionPoint
+from nncf.common.quantization.quantizer_setup import QuantizationPointBase
+from nncf.common.quantization.quantizer_setup import SingleConfigQuantizationPoint
+from nncf.common.quantization.quantizer_setup import SingleConfigQuantizerSetup
+from nncf.common.quantization.quantizer_setup import WeightQuantizationInsertionPoint
+from nncf.common.quantization.structs import QuantizationScheme as QuantizationMode
+from nncf.common.quantization.structs import QuantizerConfig
+from nncf.experimental.quantization.quantizer.quantizer import Quantizer as NNCFQuantizer
+from nncf.experimental.torch.fx.nncf_graph_builder import GraphConverter
+
+EdgeOrNode = Union[Tuple[torch.fx.Node, torch.fx.Node]]
+
+
+class TorchAOQuantizerAdapter(NNCFQuantizer):
+    """
+    Implementation of the NNCF Quantizer interface for any given torch.ao quantizer.
+    """
+
+    def __init__(self, quantizer: Quantizer):
+        self._quantizer = quantizer
+
+    def get_quantization_setup(self, model: torch.fx.GraphModule, nncf_graph: NNCFGraph) -> SingleConfigQuantizerSetup:
+        anotated_model = deepcopy(model)
+
+        # self._quantizer.transform_for_annotation is called in the nncf quantize_pt2e method
+        # before the nncf_graph building.
+        self._quantizer.annotate(anotated_model)
```

Comment:
I'm thinking about what is happened if the developer calls `get_quantization_setup` twice?

Reply:
Two separate copies of the model will be annotated due to the `deepcopy` call, two independent copy of quantizer config will be returned by the `get_quantization_setup`